.github/workflows/deploy-nightly-playground.yml

```diff
@@ -0,0 +1,105 @@
+name: Nightly Playground Validate and Deploy
+
+on:
+  schedule:
+    - cron:  '0 0 * * *'
+
+
+env:
+  DIST_VERSION: 2.12.0
+
+jobs:
+  set-os-osd-urls:
+    runs-on: ubuntu-latest
+    outputs:
+      OPENSEARCH_URL: ${{ steps.set-env.outputs.OPENSEARCH_URL }}
+      OPENSEARCH_DASHBOARDS_URL: ${{ steps.set-env.outputs.OPENSEARCH_DASHBOARDS_URL }}
+      OPENSEARCH_BUILD_ID: ${{ steps.set-env.outputs.OPENSEARCH_BUILD_ID }}
+      OPENSEARCH_DASHBOARDS_BUILD_ID: ${{ steps.set-env.outputs.OPENSEARCH_DASHBOARD_BUILD_ID }}
+      OPENSEARCH_DIST_MANIFEST_URL: ${{ steps.set-env.outputs.OPENSEARCH_DIST_MANIFEST_URL }}
+      OPENSEARCH_DASHBOARDS_DIST_MANIFEST_URL: ${{ steps.set-env.outputs.OPENSEARCH_DASHBOARDS_DIST_MANIFEST_URL }}
+    steps:
+      - name: Set Env variables
+        id: set-env
+        run: |
+          opensearch_url_latest=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${DIST_VERSION}/latest/linux/x64/tar/dist/opensearch/opensearch-${DIST_VERSION}-linux-x64.tar.gz
+          opensearch_dashboards_url_latest=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/${DIST_VERSION}/latest/linux/x64/tar/dist/opensearch-dashboards/opensearch-dashboards-${DIST_VERSION}-linux-x64.tar.gz
+          
+          opensearch_build_id=$(curl -w "%{url_effective}\n" -I -L -f -S $opensearch_url_latest -o /dev/null | cut -d / -f 8)
```

Comment:
how about reading the latest dist yml file, and parsing that using yq to fetch distribution urls, build id etc?

Reply:
Thanks for the suggestion. That works much better than what I implemented. Added the change.